jsonmodels/fields.py

```diff
@@ -21,18 +25,24 @@
 ]
 
 
-class BaseField(object):
+T = TypeVar("T")
+
+
+class BaseField(Generic[T]):
 
     """Base class for all fields."""
 
-    types = None
+    types: Tuple[Any, ...] = ()
+
+    validators: List[Any] = []
+    memory: WeakKeyDictionary
 
     def __init__(
             self,
             required=False,
             nullable=False,
             help_text=None,
-            validators=None,
+            validators: Optional[List[Any]]=None,
```

Comment:
It might be possible to improve this using Python protocols. (Validators don't have a common base class.)